src/app/store/assets/utils.ts

```diff
@@ -85,10 +92,61 @@ const finishAssetsLoading = async (
   return { slugs, balances, newMeta };
 };
 
+const finishCollectiblesLoadingWithMeta = async (data: TzktAccountAsset[]) => {
+  const slugs: string[] = [];
+  const balances: StringRecord = {};
+
+  for (const asset of data) {
+    const slug = tzktAssetToTokenSlug(asset);
+
+    slugs.push(slug);
+    balances[slug] = asset.balance;
+  }
+
+  return { slugs, balances };
+};
+
+const finishCollectiblesLoadingWithoutMeta = async (
+  data: TzktAccountAsset[],
+  knownMeta: MetadataMap,
+  chainId: string
+) => {
+  const slugsWithoutMeta = data.reduce<string[]>((acc, curr) => {
+    const slug = tzktAssetToTokenSlug(curr);
+    return knownMeta.has(slug) ? acc : acc.concat(slug);
+  }, []);
+
+  const newMetadatas = isKnownChainId(chainId)
+    ? await fetchTokensMetadata(chainId, slugsWithoutMeta).catch(err => {
+        console.error(err);
+      })
+    : null;
+
+  const slugs: string[] = [];
+  const balances: StringRecord = {};
+  const newMeta: FetchedMetadataRecord = {};
+
+  for (const asset of data) {
+    const slug = tzktAssetToTokenSlug(asset);
+
+    const metadataOfNew = newMetadatas?.[slugsWithoutMeta.indexOf(slug)];
```

Comment:
Such searches may make the app freeze while loading metadata for a lot of tokens. If the performance is fine, leave it as is; otherwise, optimize it, for example, using an object with keys of `slugsWithoutMeta` and values of `true`